reconcile/utils/slack_api.py

```diff
@@ -300,17 +308,26 @@ def _initiate_usergroups(self) -> None:
         :raises slack_sdk.errors.SlackApiError: if unsuccessful response from
         Slack API
         """
+        slack_request.labels("usergroups.list", "GET").inc()
+
         result = self._sc.usergroups_list(include_users=True)
         self.usergroups = result["usergroups"]
+        self._user_groups_initialized = True
 
     def get_usergroup(self, handle: str) -> dict[str, Any]:
+        if not self._user_groups_initialized:
+            self._initiate_usergroups()
         usergroup = [g for g in self.usergroups if g["handle"] == handle]
         if len(usergroup) != 1:
             raise UsergroupNotFoundException(handle)
         return usergroup[0]
 
     def create_usergroup(self, handle: str) -> str:
+        slack_request.labels("usergroups.create", "POST").inc()
+
         response = self._sc.usergroups_create(name=handle, handle=handle)
+        # Invalidate the usergroups list cache
+        self._user_groups_initialized = False
```

Comment:
is this response containing all info about the user group? can we add it to `self.usergroups` instead of invalidate cache to save future api calls to list usergroups in one reconcile run?